app/services/application_state_change.rb

```diff
@@ -74,6 +74,6 @@ def self.valid_states
   end
 
   def references_complete?
-    application_choice.application_form.references_complete?
+    application_choice.application_form.reload.references_complete?
```

Comment:
Nice find. This was causing me a lot of agro earlier in the week and I couldn't trace the cause.